nemoguardrails/library/injection_detection/actions.py

```diff
@@ -0,0 +1,264 @@
+        log.error(msg)
+        raise ValueError(msg)
+    injection_rules = tuple(command_injection_config.injections)
+    if not set(injection_rules) <= set(PROVIDED_MODULES):
+        # Do the easy check above first. If they provide a custom dir or a custom rules file, check the filesystem
+        if not all(
+            [
+                yara_path.joinpath(f"{module_name}.yara").is_file()
+                for module_name in injection_rules
+            ]
+        ):
+            msg = (
+                f"Provided set of `injections` in injection config {injection_rules} contains elements not in "
+                f"available rules. Provided rules are in {PROVIDED_MODULES}."
+            )
+            log.error(msg)
+            raise ValueError(msg)
+
+    return action_option, yara_path, injection_rules
+
+
+@lru_cache()
+def load_rules(yara_path: Path, rule_names: Tuple) -> Union[yara.Rules, None]:
+    """
+    Take a path to the YARA rules and a list of rule names, return the compiled rules.
+
+    Parameters
+    ----------
+    yara_path : Path to YARA rules
+    rule_names: Names of YARA rules
+
+    Returns
+    -------
+    the action option as a string
+    compiled YARA rules object
+    """
+    if len(rule_names) == 0:
+        log.warning(
+            "Injection config was provided but no modules were specified. Returning None."
+        )
+        return None
+    rules_to_load = {
+        rule_name: str(yara_path.joinpath(f"{rule_name}.yara"))
+        for rule_name in rule_names
+    }
+    try:
+        rules = yara.compile(filepaths=rules_to_load)
+    except yara.SyntaxError as e:
+        msg = f"Encountered SyntaxError: {e}"
+        log.error(msg)
+        raise e
+    return rules
+
+
+def detect_injection_mapping(result: bool) -> bool:
+    """
+    Mapping for detect_*_injection actions.
+
+    The detect_*_injection functions return True when the relevant type of data is detected, and we take the prescribed
+    action if the result is true.
+    """
+    return result
+
+
+def omit_injection(text: str, matches: list[yara.Match]) -> str:
+    """
+    Attempt to strip the offending injection attempt.
+    Note that this may not be completely effective and may still result in malicious activity.
+
+    Parameters
+    ----------
+    text : text to check for command injection
+    matches : YARA rule matches
+
+    Returns
+    -------
+    the text to check for command injection with the detected injections stripped out.
+    """
+    # Copy the text to a placeholder variable
+    modified_text = text
+    for match in matches:
+        if match.strings:
+            for match_string in match.strings:
+                for instance in match_string:
+                    if instance in modified_text:
+                        modified_text = modified_text.replace(
+                            instance.plaintext().decode("utf-8"), ""
+                        )
+    return modified_text
```

Comment:
@erickgalinkin I am not sure about the `if instance in modified_text` part, should not we check `instance.plaintext().decode("utf-8")` against `modifed_text`. I am saying this because `instance` is a YARA string object, not a string that can be directly compared with `modified_text` but I am not sure, please correct me if I'm wrong.

```suggestion
    modified_text = text
    for match in matches:
        if match.strings:
            for match_string in match.strings:
                for instance in match_string:
                    try:
                        plaintext = instance.plaintext().decode("utf-8")
                        if plaintext in modified_text:
                            modified_text = modified_text.replace(plaintext, "")
                    except (AttributeError, UnicodeDecodeError) as e:
                        log.warning(f"Error processing match: {e}")
    return modified_text

```

Reply:
applied as part of https://github.com/NVIDIA/NeMo-Guardrails/commit/af80ad7f4d236ca39abf8d8d45d03a9830d2936a

please see https://github.com/NVIDIA/NeMo-Guardrails/pull/1091#issuecomment-2789468118